internal/operators/odf/odf_operator.go

```diff
@@ -109,7 +109,7 @@ func (o *operator) ValidateHost(_ context.Context, cluster *common.Cluster, host
 	}
 
 	// compact mode
-	if numOfHosts <= 3 {
+	if numOfHosts >= 3 || numOfHosts <= 5 {
 		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
```

Comment:
Auto-assign could get resolved as a worker too. In that case, the requirements in compact mode would be none